packages/eslint-config-typescript/parser-overrides/cypress.js

```diff
@@ -0,0 +1,26 @@
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Determines if the project has a `<root>/cypress/tsconfig.json` file
+ *
+ * @returns {boolean}
+ */
+function hasCypressTSConfig() {
+  const tsConfigPath = path.join(process.cwd(), 'cypress', 'tsconfig.json');
+  return fs.existsSync(tsConfigPath);
+}
+
+module.exports = {
+  overrides: [
+    // Adds the additional tsconfig.json to the `@typescript-eslint/parser` configuration
+    hasCypressTSConfig()
```

Comment:
Crafty! Did it throw errors to have the override config just at the top level if the cypress/tsconfig.json didn't exist?

Makes sense since the override is for the whole cypress folder just interested to hear what the error was

Reply:
It did throw an error: since the main tsconfig excludes `cypress/`, the parser would fail with a non-zero exit code upon trying to lint that directory. By adding in this override to register the additional tsconfig.json for `cypress/`, the error is resolved and linting works as expected.

I added a bit of an explainer in https://github.com/reside-eng/lint-config/blob/7b68bb48af74ee26aaad46c502eb87c90e8f5d62/packages/eslint-config-typescript/README.md#multiple-tsconfigjson-files if you want to know more!